File: java/src/main/java/com/example/livecoding/recursion/RecursiveFibonacci.java
package com.example.livecoding.recursion;

public final class RecursiveFibonacci {

    private RecursiveFibonacci() {
    }

    public static long recursiveFibonacci(int n) {
        if (n < 0) {
            throw new IllegalArgumentException("n must be non-negative");
        }
        if (n <= 1) {
            return n;
        }
        return recursiveFibonacci(n - 1) + recursiveFibonacci(n - 2);
    }
}

File: python/src/recursion/recursive_fibonacci.py
"""Fibonacci F(0)=0, F(1)=1; naive recursive (for practice only)."""


def recursive_fibonacci(n: int) -> int:
    if n < 0:
        raise ValueError("n must be non-negative")
    if n <= 1:
        return n
    return recursive_fibonacci(n - 1) + recursive_fibonacci(n - 2)
